src/metrics/tracking.ts

```diff
@@ -0,0 +1,99 @@
+import * as amplitude from "@amplitude/analytics-browser";
+import { AnyObject } from "@/types/types";
+
+export const initTracking = () => {
+  if (!process.env.NEXT_PUBLIC_AMPLITUDE_API_KEY) {
+    return;
+  }
+
+  const options = {
+    defaultTracking: false,
+    attribution: false,
+    pageViews: true,
+    sessions: true,
+    formInteractions: false,
+    fileDownloads: false,
+    logLevel:
+      process.env.NODE_ENV === "development"
+        ? amplitude.Types.LogLevel.Warn
+        : amplitude.Types.LogLevel.None,
+  };
+
+  try {
+    amplitude.init(process.env.NEXT_PUBLIC_AMPLITUDE_API_KEY, options);
+  } catch (e) {
+    // Do nothing
+  }
+};
+
+export const trackEvent = (type: TrackingEvent, properties?: AnyObject) => {
+  if (!process.env.NEXT_PUBLIC_AMPLITUDE_API_KEY) {
+    return;
+  }
+
+  try {
+    amplitude.track(type, properties);
+  } catch (e) {
+    // Do nothing
+  }
+};
+
+export enum TrackingEvent {
```

Comment:
@janewang These are some basic events we can start with, but we can always tweak them and add more.

Reply:
@quietbits Looks great as a starting point, thank you